Alg.java

```diff
@@ -0,0 +1,605 @@
+import java.util.PriorityQueue;
+import java.lang.Integer;
+import java.util.ArrayList;
+
+public class Alg {
```

Comment:
Let's change this to "Search.java" 
We can have this class be responsible for all things search related.